### PR TITLE
Check `IsActionsToken` for LFS authentication

### LINKED ISSUE #23824: Gitea Actions fetching from lfs, api error: Authentication required: Unauthorized
### Description

Hi,

I use the new gitea actions feature with a private repo which has lfs usage.

Please note, that I use another docker image because default debian image from your default act_runner config does not have "git lfs" and "yarn" installed.

My testing workflow:

```yaml
name: deploy to production

on: "push"

jobs:
    deploy:
        name: deploy
        runs-on: ubuntu-latest
        container:
            image: catthehacker/ubuntu:js-latest
        steps:
            - uses: actions/checkout@v3
              with:
                  lfs: true
```

I think using the other image should not be an issue. I am using your default `actions/checkout@v3`.

Here is the log of gitea actions web ui:

```
Resolving deltas:  50% (1/2)
Resolving deltas: 100% (2/2)
Resolving deltas: 100% (2/2), done.
From https://*********
 * [new ref]         4e3676277fe8441ac162714068eb284087a0bbf3 -> origin/master
::endgroup::
::group::Determining the checkout info
::endgroup::
::group::Fetching LFS objects
[command]/usr/bin/git lfs fetch origin refs/remotes/origin/master
fetch: Fetching reference refs/remotes/origin/master
batch response: Authentication required: Unauthorized
batch response: Authentication required: Unauthorized
batch response: Authentication required: Unauthorized
batch response: Authentication required: Unauthorized
batch response: Authentication required: Unauthorized
error: failed to fetch some objects from 'https://**********.git/info/lfs'
The process '/usr/bin/git' failed with exit code 2
Waiting 13 seconds before trying again
```

I removed the first successfull lines of the log (successfully fetching) and the server url, because it is a private server.

The act runner is built today.

I did not test it on try.gitea.io, because actions is not enabled in this environment.

Thank you for your help and this great tool

### Gitea Version

1.19.0

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

linux

### How are you running Gitea?

Gitea runs in docker.

### Database

SQLite

### PR DESCRIPTION
**Pull request by Zettat123**
Close #23824 

Actions cannot fetch LFS objects from private repos because we don't check if the user is the `ActionUser`.

### CODE REVIEW

File: services/lfs/server.go [hunk 1]
```diff
@@ -495,25 +496,44 @@ func authenticate(ctx *context.Context, repository *repo_model.Repository, autho
 		accessMode = perm.AccessModeWrite
 	}
 
-	// ctx.IsSigned is unnecessary here, this will be checked in perm.CanAccess
-	perm, err := access_model.GetUserRepoPermission(ctx, repository, ctx.Doer)
-	if err != nil {
-		log.Error("Unable to GetUserRepoPermission for user %-v in repo %-v Error: %v", ctx.Doer, repository)
-		return false
-	}
+	if ctx.Data["IsActionsToken"] == true {
```
Comment (yp05327) [inline services/lfs/server.go:499]:
```suggestion
	if ctx.Doer.IsActions() {
```

File: services/lfs/server.go [hunk 2]
```diff
@@ -495,25 +496,44 @@ func authenticate(ctx *context.Context, repository *repo_model.Repository, autho
...
 
-	canRead := perm.CanAccess(accessMode, unit.TypeCode)
-	if canRead && (!requireSigned || ctx.IsSigned) {
-		return true
-	}
+		if task.IsForkPullRequest {
+			return accessMode <= perm.AccessModeRead
+		} else {
+			return accessMode <= perm.AccessModeWrite
+		}
```
Comment (yp05327) [inline services/lfs/server.go]:
```suggestion
		return accessMode <= perm.AccessModeWrite
```
Maybe lint error here.

File: services/lfs/server.go [hunk 3]
```diff
@@ -495,25 +496,44 @@ func authenticate(ctx *context.Context, repository *repo_model.Repository, autho
...
+	} else {
```
Comment (yp05327) [inline services/lfs/server.go]: Convert `ctx.Doer` into `TriggerUser`?
Not sure whether it is necessary here.

File: services/lfs/server.go (hunk 1 above)
Comment (yp05327) [reply services/lfs/server.go:499]: emmm, we need to check `ctx.Doer == nil` if use `ctx.Doer.IsActions()`
maybe the old code is better.

Comment (Zettat123) [reply services/lfs/server.go:499]: The `IsActions` func will check if user is `nil`, so I think using `ctx.Doer.IsActions()` is ok
https://github.com/go-gitea/gitea/blob/ca03ca9e6e8f2e7ba7fdc3a64edb766be476edbd/models/user/user_system.go#L62-L64

File: services/lfs/server.go (hunk 3 above)
Comment (Zettat123) [reply services/lfs/server.go:514]: The code in `else` block is the old code before this change. This PR is to fix the authentication for `ActionUser` so it's better to keep the old code unchanged.

Comment (yp05327) [reply services/lfs/server.go]: Ah, we need to convert ActionUser into TriggerUser and check the TriggerUser’s permission.
Same as #23729

**Comment by wolfogre** [comment]
It follows the login in https://github.com/go-gitea/gitea/blob/1bc4ffc33739c661891278c3987c16bf63142ad6/routers/web/repo/http.go#L198.

File: services/lfs/server.go (hunk 3 above)
Comment (Zettat123) [reply services/lfs/server.go]: > Convert `ctx.Doer` into `TriggerUser`? 

I think it's not necessary. LFS is a part of a repo, so the workflow of a repo should be able to access its LFS objects and we don't need to convert the user and check permission again.

**Review by wolfogre — APPROVED** [review]

File: services/lfs/server.go [hunk 4]
```diff
@@ -495,10 +496,27 @@ func authenticate(ctx *context.Context, repository *repo_model.Repository, autho
 		accessMode = perm.AccessModeWrite
 	}
 
+	if ctx.Doer.IsActions() {
```
Comment (lunny) [inline services/lfs/server.go]: You cannot assume `ctx.Doer` is not `nil`. I still think the previous reversion is better.

**Review by lunny — CHANGES_REQUESTED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**